app/models/day.rb

```diff
@@ -35,37 +35,50 @@ class Day < ApplicationRecord
              foreign_key: :day_of_week,
              primary_key: :day
 
+  has_many :non_working_days,
+           inverse_of: false,
+           class_name: 'NonWorkingDay',
+           foreign_key: :date,
+           primary_key: :date
+
   attribute :date, :date, default: nil
   attribute :day_of_week, :integer, default: nil
-  attribute :working, :boolean, default: 't'
 
-  delegate :name, to: :week_day
+  delegate :name, to: :week_day, allow_nil: true
 
-  def self.default
+  def self.default_scope
     today = Time.zone.today
     from = today.at_beginning_of_month
     to = today.next_month.at_end_of_month
 
-    select('days.*')
-      .includes(:week_day)
-      .from(Arel.sql(from_sql(from:, to:)))
+    from(Arel.sql(from_sql(from:, to:)))
+    .includes(:week_day)
+    .includes(:non_working_days)
+    .order(:date)
```

Comment:
The removal of the `working` property leads to the equally named filter to no longer work. E.g. a request for 

```
http://localhost:3000/api/v3/days?filters=[{"date": {"operator": "<>d", "values": ["2022-05-01", "2022-05-07"] } }, { "working": { "operator": "=", "values": ["t"] } }]
```

will fail now.

The calculation of whether a day is a working day or not thus also needs to happen inside sql.

To avoid a regression such as this we should add a [request spec for the `DaysAPI`](https://github.com/opf/openproject/blob/dev/spec/requests/api/v3/days/day_spec.rb)